core/runtime/src/main/java/io/quarkus/runtime/configuration/StaticInitSafe.java

```diff
@@ -0,0 +1,22 @@
+package io.quarkus.runtime.configuration;
+
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Used to mark a ConfigSource, ConfigSourceProvider or ConfigSourceFactory as safe to be initialized during STATIC
```

Comment:
I think we need to explain what static-init is here, as I am sure a lot of users won't know it. Also having a small example in the javadoc would be extremely useful I think

Reply:
Sure, I'll clarify.